extensions/cornerstone/src/services/SegmentationService/SegmentationService.ts

```diff
@@ -448,53 +452,50 @@ class SegmentationService {
 
       if (!suppressEvents) {
         this._broadcastEvent(this.EVENTS.SEGMENTATION_UPDATED, {
-          segmentation,
+          segmentation: cachedSegmentation,
         });
       }
 
       return segmentationId;
     }
 
-    // Add the segmentation otherwise
+    const representationType = segmentation.type;
+    const representationData =
+      segmentation.representationData[representationType];
     cstSegmentation.addSegmentations([
       {
         segmentationId,
         representation: {
-          type: LABELMAP,
-          // Todo: need to be generalized
+          type: representationType,
           data: {
-            volumeId: segmentationId,
+            ...representationData,
           },
         },
       },
     ]);
 
     // Define a new color LUT and associate it with this segmentation.
-
     // Todo: need to be generalized to accept custom color LUTs
     const newColorLUT = this.generateNewColorLUT();
     const newColorLUTIndex = this.getNextColorLUTIndex();
 
     cstSegmentation.config.color.addColorLUT(newColorLUT, newColorLUTIndex);
 
-    if (
-      segmentationSchema.label === undefined ||
-      segmentationSchema.label === ''
-    ) {
-      segmentationSchema.label = 'Segmentation';
-    }
-
     this.segmentations[segmentationId] = {
-      ...segmentationSchema,
-      segments: segmentationSchema.segments || [null],
-      activeSegmentIndex: segmentationSchema.activeSegmentIndex ?? null,
-      segmentCount: segmentationSchema.segmentCount ?? 0,
+      ...segmentation,
+      label:
+        segmentation.label !== undefined || segmentation.label !== ''
+          ? segmentation.label
+          : 'Segmentation',
```

Comment:
I'd rather see the empty string here than a hard coded Segmentation label.  Then, on display, the display component can decide how to handle it, and it becomes:
segmentation.label ||= '';
which is quite a bit easier to read.